src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -49,8 +48,6 @@ def self.initialize_puppet(puppet_config)
     Puppet::FileServing::Metadata.indirection.terminus_class = :file_server
     Puppet::FileBucket::File.indirection.terminus_class = :file
 
-    Puppet::Node.indirection.cache_class = Puppet[:node_cache_terminus]
```

Comment:
Discussing with @joshcooper it would be better to have left this in place - honoring the setting value would make it easier for the user to keep the 4.x behavior if they wanted (as opposed to having to configure this with a `routes.yaml` entry for node. I'm going to open a separate PR.